### PR TITLE
Issue #1326 - Rollback UI is not showing correct ksonnet parameters in preview

### PR DESCRIPTION
**Pull request by alexmt**

### CODE REVIEW

File: util/ksonnet/ksonnet.go [hunk 1]
```diff
@@ -158,18 +162,39 @@ func (k *ksonnetApp) ListParams() ([]*v1alpha1.KsonnetParameter, error) {
 	if err := json.Unmarshal([]byte(out), &ksParams); err != nil {
 		return nil, err
 	}
-	var params []*v1alpha1.KsonnetParameter
+	params := make(map[string]*v1alpha1.KsonnetParameter)
 	for _, ksParam := range ksParams.Data {
 		value := strings.Trim(ksParam.Value, `'"`)
-		params = append(params, &v1alpha1.KsonnetParameter{
+		params[fmt.Sprintf("%s/%s", ksParam.Component, ksParam.Key)] = &v1alpha1.KsonnetParameter{
```
Comment (alexec) [inline util/ksonnet/ksonnet.go]: Minor, but I feel like we should have `KsonnetParameter.String()`  or similar.

**Review by alexec — APPROVED** [review]

File: util/ksonnet/ksonnet.go (hunk 1 above)
Comment (alexmt) [reply util/ksonnet/ksonnet.go]: KsonnetParameter includes Value in addition to Component and Key. I would prefer to keep it as is.

File: util/ksonnet/ksonnet.go [hunk 2]
```diff
@@ -158,18 +162,39 @@ func (k *ksonnetApp) ListParams() ([]*v1alpha1.KsonnetParameter, error) {
...
 			Name:      ksParam.Key,
 			Value:     value,
-		})
+		}
 	}
 	return params, nil
 }
 
+func (k *ksonnetApp) ListParams(environment string) ([]*v1alpha1.KsonnetParameter, error) {
+	params, err := k.listParams("")
```
Comment (jessesuen) [inline util/ksonnet/ksonnet.go]: I don't believe it is necessary to list params first without environment specified, and then list params with environment. Ksonnet will give a complete list. Did you find a reason you needed to do this?

**Comment by alexmt** [comment]
I think I hit ksonnet bug and decided that this is the expected behavior. This is why unit test failed. Will the test app and change the ListParams to list params only once.

File: util/ksonnet/ksonnet.go (hunk 2 above)
Comment (alexmt) [reply util/ksonnet/ksonnet.go]: fixed

**Pull request merged**